ext/uri/php_uri.c

```diff
@@ -337,18 +334,17 @@ ZEND_ATTRIBUTE_NONNULL_ARGS(1, 2) PHPAPI void php_uri_instantiate_uri(
 		uri_object = Z_URI_OBJECT_P(return_value);
 	}
 
-	const php_uri_parser *uri_parser = uri_object->internal.parser;
+	const php_uri_parser *uri_parser = uri_object->parser;
 
 	zval errors;
 	ZVAL_UNDEF(&errors);
 
 	void *base_url = NULL;
 	if (base_url_object != NULL) {
 		ZEND_ASSERT(base_url_object->std.ce == uri_object->std.ce);
-		const uri_internal_t *internal_base_url = &base_url_object->internal;
-		URI_ASSERT_INITIALIZATION(internal_base_url);
-		ZEND_ASSERT(internal_base_url->parser == uri_parser);
-		base_url = internal_base_url->uri;
+		ZEND_ASSERT(base_url_object->uri != NULL);
+		ZEND_ASSERT(base_url_object->parser == uri_parser);
+		base_url = base_url_object->uri;
```

Comment:
I think the improvement is particularly visible here, since the assertions are no longer mixed with declarations.